### ISSUE TITLE
Upgrade from Rails 3.2 to Rails 4.2.8

### ISSUE THREAD

**Issue by sashadev-sky** [publiclab/mapknitter#481]
References #305 #345 

Current system versions:
Rails 3.2
Ruby 2.4.4

**Comment by sashadev-sky** [comment]
Tagging some relevant people @cesswairimu @Souravirus @jywarren @stefannibrasil @gauravano

**Comment by kaunair** [comment]
Rails 4.0 and Ruby 2.4 are not compatible. ( Relevant issue: https://github.com/rails/rails/issues/27670)
You'll run into `<class:Numeric>': stack level too deep (SystemStackError)` error
The best approach is to directly upgrade to Rails 4.2.8 or higher
Thanks

**Comment by sashadev-sky** [comment]
@kaunair sorry i meant to tag you in this too!! I'm actually not going to be able to open a PR for this for a week or so, so you can address it if you wanted to get a head start! 

Thank you for advising me on that though I will keep that in mind if I do end up opening the PR for it. @alaxalves mentioned he went with 4.2.8 so I'm going to run with that too as per your suggestion. Updating the issue name! Thanks!

**Comment by alaxalves** [comment]
@sashadev-sky I have opened a PR regarding this in https://github.com/alaxalves/mapknitter/tree/rails4, as I said [here](https://github.com/publiclab/mapknitter/pull/363#issuecomment-470319037) but @jywarren told me to wait for it a little [here](https://github.com/publiclab/mapknitter/pull/363#issuecomment-470319576).

**Comment by sashadev-sky** [comment]
@alaxalves Yeah just getting the ball rolling. There will still be a lot of work to do for SoC in terms of upgrades even if we get it to Rails 4. But @kaustubh-nair had a proposal for doing the upgrades and he's getting involved here early (thank you!)

**Comment by sashadev-sky** [comment]
@alaxalves Sorry I just realized you might have a proposal for this too based on a comment you wrote!! I'm not involved in the proposal process at all I just happened to see @kaustubh-nair's at some point so apologies if I created any confusion or left you out in the last comment. Just trying to upgrade Rails sooner and thank you for helping too :)

**Comment by gauravano** [comment]
I think you all can divide the upgrade tasks and work keeping each other in sync?

**Comment by kaustubh-nair** [comment]
The upgrade is paused for now. Since we don't have enough test coverage. We'll continue after that

**Comment by sashadev-sky** [comment]
@kaustubh-nair yeah thats what I think @gauravano was getting at - dividing up the test coverage and other work between us

**Comment by sashadev-sky** [comment]
+1 @gauravano

**Comment by gauravano** [comment]
Also, you can get expert help from @Souravirus (expert of big PRs :see_no_evil: ) and he also did this kind of upgrade last year for plots2 - https://github.com/jywarren/plots2/pull/266 and then Rails upgrade in GSoC. And, you know @siaw23 (the Rubyist)

Thanks!

**Comment by cesswairimu** [comment]
Just noting here, I was trying to build on top of #482 as we had discussed with @kaustubh-nair but was finding it overwhelming because of the different changes in the in between versions. I also found elsewhere that the Rails 4.0 and Ruby 2.4 compatibility was fixed. Trying out if this will work on https://github.com/publiclab/mapknitter/pull/578. This would be better if it works for debugging and making sure we don't miss anything between the versions. Also if it doesn't work will fall back to #482.
